### ISSUE TITLE
why chunked compression rate and how to interpret It? 

### ISSUE THREAD

**Issue by MatteoMichetti01** [google-deepmind/language_modeling_is_compression#18]
"Why is the compression with the language model only chunked? I followed all the steps and trained the model for 1 million steps, obtaining a high loss (2552.305908). I compressed enwik9 (using python compress.py --compressor=language_model), and after 35 hours, I got 'Chunked: 23.5'. How should I interpret this?"

**Comment by anianruoss** [comment]
**Why do I only obtain a chunked compression ratio for transformers?**

Unless you find a way to train a transformer with a context size of a billion (given that the dataset is 1GB and every token is a byte), you will have to chunk the data to fit it into the context window.

**Why is the loss high?**

The code only provides an example configuration for illustration purposes. If you want to achieve competitive performance, you will have to tune the hyperparameters.

**Comment by MatteoMichetti01** [comment]
"I followed all the suggested tuning except for the implementation of 'rotary' for the 200k transformer. Do you think the result I obtained (chunked: 23.5) is not valid, considering that in the paper they reach 30.9?

**Comment by anianruoss** [comment]
The lowest compression ratio in the paper is 17.7, so I don’t see why your result shouldn’t be valid.

**Comment by MatteoMichetti01** [comment]
Because I thought that by training the transformer with the hyperparameter configuration, I would have obtained the same result as in the paper, which is 30.9. Thank you very much for the thorough answers. I wanted to ask one last question: to train the transformer on smaller versions of enwik9 like enwik7 or enwik6, is it enough to modify: data_generator = data_loaders.get_enwik9_iterator(
        num_chunks=constants.NUM_CHUNKS // 100 or 1000? in
```"
train.py?

**Comment by anianruoss** [comment]
**Why are the results different?**

You mentioned that you did not use the rotary positional encodings, so I’m not surprised that the results are not entirely the same. 

**How do I train on less data?**

Yes, reducing the number of chunks will give you a smaller dataset (if you keep the sequence length constant).

**Comment by MatteoMichetti01** [comment]
> **Why are the results different?**
> 
> You mentioned that you did not use the rotary positional encodings, so I’m not surprised that the results are not entirely the same.
> 
> **How do I train on less data?**
> 
> Yes, reducing the number of chunks will give you a smaller dataset (if you keep the sequence length constant).

**Comment by MatteoMichetti01** [comment]
Do I also need to change these hyperparameters to adapt them to the smaller dataset?

"training_steps": "1000000", "batch_size": "32", "seq_length": "2048", "embedding_dim": "64", "num_heads": "4", "num_layers": "4",

Do you have any suggestions, like halving them, for example? Thank you

**Comment by anianruoss** [comment]
You will most likely have to adapt the hyperparameters to the smaller datasets, but, unfortunately, I cannot provide any general guidance on how to do so.

**Comment by MatteoMichetti01** [comment]
Could it make sense to reduce the sequence length and increase the number of chunks to reduce the loss on enwik6? For example, with a sequence length of 64 bytes, which is 32 times smaller than the default size, increase the number of chunks by 32 times so as to cover the same number of dataset bytes

**Comment by anianruoss** [comment]
Figure 4 in the paper shows the impact of the sequence length on the compression ratio. As you can see, a longer sequence length, generally leads to better performance. Nevertheless, you will have to ablate this factor empirically.

**Comment by MatteoMichetti01** [comment]
Just to be clear, is the output of compress.py the adjusted compression rate?

**Comment by anianruoss** [comment]
No, it's the unadjusted compression rate (see, e.g., https://github.com/google-deepmind/language_modeling_is_compression/blob/b5c8f8a63349d0a2604367d47df4a7c79db52890/compress.py#L138).
